### PR TITLE
Introduced tests for `repository-packages`

### PR DESCRIPTION
**Pull request by edynox**
Unit tests for these features were removed in [swdb PR](https://github.com/rpm-software-management/dnf/pull/630) due to incompatibility with swdb.

### CODE REVIEW

File: dnf-docker-test/features/steps/command_steps.py [hunk 1]
```diff
@@ -59,6 +59,23 @@ def step_the_command_exit_code_is(ctx, values):
             codes.append(int(part.strip()))  # just plain number
     ctx.assertion.assertIn(ctx.cmd_result.returncode, codes)
 
+@then("the command {stream:stdout_stderr} should match")
+def step_the_command_stream_should_match(ctx, stream):
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/steps/command_steps.py]: no documentation

File: dnf-docker-test/features/repository-packages-info.feature [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            Summary      : Empty
+            License      : Public Domain
+            Description  : Empty.
+
+            Available Packages
+            Name         : TestB
+            Version      : 1
+            Release      : 1
+            Arch         : noarch
+            Size         : 5.4 k
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/repository-packages-info.feature]: meh

**Comment by edynox** [comment]
@ignatenkobrain 
fixed indent and edited doc.

File: README.md [hunk 3]
```diff
@@ -102,13 +102,13 @@ Here's an example configuration from the first ported test:
...
  TestA requires (TestB or TestC), TestA recommends TestC
 
-Scenario: Install TestA from repository "test-1"
- Given I use the repository "test-1"
- When I execute "dnf" command "install -y TestA" with "success"
- Then transaction changes are as follows
-   | State        | Packages      |
-   | installed    | TestA, TestC  |
-   | absent       | TestB         |
+    Scenario: Install TestA from repository "test-1"
```
Comment (j-mracek) [inline README.md]: Please can you rewrite the scenario into new syntax. Like you use here:
```
Scenario: Sync package to latest version in available repositories
+         When I save rpmdb
+          And I enable repository "distro2"
+          And I successfully run "dnf -y repository-packages distro1 remove-or-distro-sync TestA"
+         Then rpmdb changes are
+           | State    | Packages |
+           | upgraded | TestA    |
```

**Comment by j-mracek** [comment]
I have found that test repository-packages-remove is failing (after step: ```And I successfully run "dnf -y repository-packages test2 remove-or-reinstall TestC"```)

**Comment by j-mracek** [comment]
@edynox Please can you investigate the failing test?

**Comment by edynox** [comment]
@j-mracek I fixed that Readme...

I know about that issue and I was trying to fix it in some other test, but I'm not familiar with RPM API.
Problem is in comparing RPMDB changes here 
https://github.com/rpm-software-management/ci-dnf-stack/blob/master/dnf-docker-test/features/steps/rpm_utils.py#L101

Under some circumstances rpmdb header doesn't change after the reinstallation. There is some timestamp so you can workaround it by waiting a second, but sometimes it doesn't help.

I got recommendation that looking at the number of the rpmdb entry might work.
```
see rpm.mi.instance()
the mi aka Matchterator gets returned from dbMatch()
```

I tried something like:
```
        a = pre.dsOfHeader()
        b = post.dsOfHeader()
        if pre.unload() == post.unload() and a.Instance() == b.Instance():
            return State.unchanged
        else:
            return State.reinstalled
```
But it doesn't work....
Any ideas? @ignatenkobrain 

From output you can see that package was reinstalled
```
      | State       | Packages |
      | reinstalled | TestC    |
      Assertion Failed: u'TestC-1-1.noarch' -> u'TestC-1-1.noarch'

shell.command.stdout:

      Last metadata expiration check: 0:00:00 ago on Fri Mar 31 11:59:49 2017 UTC.
      Dependencies resolved.
      ================================================================================
       Package           Arch               Version           Repository         Size
      ================================================================================
      Reinstalling:
       TestC             noarch             1-1               test4             5.4 k
      
      Transaction Summary
      ================================================================================
      
      Total size: 5.4 k
      Running transaction check
      Transaction check succeeded.
      Running transaction test
      Transaction test succeeded.
      Running transaction
        Reinstalling: TestC-1-1.noarch                                            1/2 
        Erasing     : TestC-1-1.noarch                                            2/2 
        Verifying   : TestC-1-1.noarch                                            1/2 
        Verifying   : TestC-1-1.noarch                                            2/2 
      
      Reinstalled:
        TestC.noarch 1-1                                                              
      
      Complete!
```

**Comment by j-mracek** [comment]
There could be work around, instead of test if package was reinstalled, you can test output of transaction if in transaction table are correct elements. If rpm store install-time with second as minimal difference, we cannot do much.

**Comment by edynox** [comment]
Well, I can just use
```
 Then history info should match
        | Key         | Value  |
        | Reinstall   | TestC  |
```
(after history tests will be merged)

But not sure if it is reliable.

Also this problem appears in every test using `reinstall` with `rpmdb changes are`

**Comment by j-mracek** [comment]
@edynox I will investigate the problem here

**Comment by j-mracek** [comment]
Thanks, it looks fine and it works

**Pull request merged**